Code/GraphMol/RascalMCES/RascalMCES.cpp

```diff
@@ -1126,15 +1126,25 @@ std::vector<RascalResult> findMCES(RascalStartPoint &starter,
                      opts.maxFragSeparation, opts.exactConnectionsMatch,
                      opts.equivalentAtoms, opts.ignoreBondOrders));
   }
+  // If 2 cliques are the same size, this sort puts the one with the smaller
+  // number of fragments first, which may have fewer atoms.
+  std::sort(results.begin(), results.end(), details::resultCompare);
 
   if (opts.singleLargestFrag) {
-    std::sort(results.begin(), results.end(),
-              [](const RascalResult &r1, const RascalResult &r2) -> bool {
-                if (r1.getAtomMatches() == r2.getAtomMatches()) {
-                  return (r1.getBondMatches() < r2.getBondMatches());
-                }
-                return (r1.getAtomMatches() < r2.getAtomMatches());
-              });
+    std::sort(
+        results.begin(), results.end(),
+        [](const RascalResult &r1, const RascalResult &r2) -> bool {
+          if (r1.getAtomMatches().size() == r2.getAtomMatches().size()) {
+            if (r1.getBondMatches().size() == r2.getBondMatches().size()) {
+              if (r1.getAtomMatches() == r2.getAtomMatches()) {
+                return (r1.getBondMatches() < r2.getBondMatches());
+              }
+              return r1.getAtomMatches() < r2.getAtomMatches();
+            }
+            return r1.getBondMatches().size() > r2.getBondMatches().size();
+          }
+          return (r1.getAtomMatches().size() > r2.getAtomMatches().size());
```

Comment:
why do the first two return statements here return r1 < r2 and the second two return r1 > r2?

Reply:
It's sorting in descending number of bonds and atoms, ascending order of atom and bond indices.  The sort on vector size is important, on index less so but I prefer lower indices first for no good reason.